process/block/shardblock.go

```diff
@@ -1637,18 +1637,26 @@ func (sp *shardProcessor) ApplyBodyToHeader(hdr data.HeaderHandler, bodyHandler
 		go sp.checkAndRequestIfMetaHeadersMissing(hdr.GetRound())
 	}()
 
-	if bodyHandler == nil || bodyHandler.IsInterfaceNil() {
-		return nil
+	if check.IfNil(bodyHandler) {
+		return bodyHandler, nil
```

Comment:
we are returning nil, nil here. Why not returning an error?

Reply:
done.